### ISSUE TITLE
False negatives on the `Style/RedundantLineContinuation` cop

### ISSUE THREAD

**Issue by ydakuka** [rubocop/rubocop#12430]
## Describe the solution you'd like

```ruby
def create
  # TODO: some code

  if @user.confirmed?
    redirect_to user_path(@user),
                notice: translate('confirmed') and return
  elsif @user.condition?
    redirect_to user_path(@user), \
                notice: translate('not_confirmed') and return
  end

  # TODO: some code
end
```

```ruby
let! :user do
  create(:user,
         my_foo: my_foo,
         my_bar: my_bar, \
         my_barbaz: my_barbaz)
end
```

```ruby
let! :user do
  create(:user, \
         my_foo: my_foo, \
         my_bar: my_bar,
         my_baz: my_baz, \
         my_foobar: my_foobar)
end
```

## Rubocop

```console
ydakuka@yauhenid:~/Work/project$ bin/rails_docker rubocop -V
1.58.0 (using Parser 3.2.2.4, rubocop-ast 1.30.0, running on ruby 2.7.8) [x86_64-linux]
  - rubocop-capybara 2.19.0
  - rubocop-factory_bot 2.24.0
  - rubocop-performance 1.19.1
  - rubocop-rails 2.22.2
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.25.0
  - rubocop-thread_safety 0.5.1
```

**Comment by bkuhlmann** [comment]
To add to the above, I'm seeing this issue crop up in the RuboCop 1.61.0 release for the first time as well but in the specs. Example:

``` 
expect(temp_dir.join("test/Procfile.dev").read).to eq(
  %(web: rerun --dir app,config,lib,slices --pattern="**/*.{erb,rb}" ) \  # <-- Flagged, incorrectly, as redundant line continuation
  "-- bundle exec puma --config ./config/puma.rb\n" \
  "assets: bundle exec hanami assets watch\n"
)
```

<details>
  <summary>RuboCop Version Details</summary>
  
``` 
1.61.0 (using Parser 3.3.0.5, rubocop-ast 1.31.1, running on ruby 3.3.0) [arm64-darwin23.3.0]
  - rubocop-capybara 2.20.0
  - rubocop-factory_bot 2.25.1
  - rubocop-packaging 0.5.2
  - rubocop-performance 1.20.2
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.27.0
  - rubocop-thread_safety 0.5.1
```
</details>

**Comment by ydakuka** [comment]
https://github.com/rubocop/rubocop/issues/12862

**Comment by akimd** [comment]
I'm facing the same issue.

```
$ cat backslash.rb
puts foo(123,
         123) \
     == bar
```
```
$ bundle exec rubocop backslash.rb
Inspecting 1 file
C

Offenses:

backslash.rb:1:1: C: [Correctable] Style/FrozenStringLiteralComment: Missing frozen string literal comment.
puts foo(123,
^
backslash.rb:2:15: C: [Correctable] Style/RedundantLineContinuation: Redundant line continuation.
         123) \ ...
              ^

1 file inspected, 2 offenses detected, 2 offenses autocorrectable
```

**Comment by Earlopain** [comment]
@akimd can you make a new issue? This is about code that should register an offense. I think your problem is about the opposite?

**Comment by akimd** [comment]
Bummer, I read "false positive", but it's about false negatives, you are right. Sorry.

I opened https://github.com/rubocop/rubocop/issues/13145.

**Comment by dvandersluis** [comment]
@ydakuka's false negative examples (https://github.com/rubocop/rubocop/issues/12430#issue-2021021437) and @bkuhlmann's false positive examples (https://github.com/rubocop/rubocop/issues/12430#issuecomment-1974810097) are all fixed now so I'm going to close this issue.

**Comment by ydakuka** [comment]
@dvandersluis I checked my false negative examples, and except for the first one, they have not been fixed yet.

**Comment by dvandersluis** [comment]
@ydakuka what version of rubocop are you using? I tried this on master:

```ruby
# test.rb
# frozen_string_literal: true

def create
  # TODO: some code

  if @user.confirmed?
    redirect_to user_path(@user),
      notice: translate('confirmed') and return
  elsif @user.condition?
    redirect_to user_path(@user), \
      notice: translate('not_confirmed') and return
  end

  # TODO: some code
end

let! :user do
  create(:user,
    my_foo: my_foo,
    my_bar: my_bar, \
    my_barbaz: my_barbaz)
end

let! :user do
  create(:user, \
    my_foo: my_foo, \
    my_bar: my_bar,
    my_baz: my_baz, \
    my_foobar: my_foobar)
end
```

```
$ rubocop test.rb --only Style/RedundantLineContinuation
Inspecting 1 file
C

Offenses:

test.rb:10:35: C: [Correctable] Style/RedundantLineContinuation: Redundant line continuation.
    redirect_to user_path(@user), \ ...
                                  ^
test.rb:20:21: C: [Correctable] Style/RedundantLineContinuation: Redundant line continuation.
    my_bar: my_bar, \ ...
                    ^
test.rb:25:17: C: [Correctable] Style/RedundantLineContinuation: Redundant line continuation.
  create(:user, \ ...
                ^
test.rb:26:21: C: [Correctable] Style/RedundantLineContinuation: Redundant line continuation.
    my_foo: my_foo, \ ...
                    ^
test.rb:28:21: C: [Correctable] Style/RedundantLineContinuation: Redundant line continuation.
    my_baz: my_baz, \ ...
                    ^

1 file inspected, 5 offenses detected, 5 offenses autocorrectable
```

**Comment by dvandersluis** [comment]
@ydakuka thank you for the examples. I have replicated the false negatives now, and opened #13609 to fix them.

**Comment by ydakuka** [comment]
@dvandersluis 

I updated rubocop to version `1.70.0` in my project, fixed all offenses, and found more false negatives.

1.

```ruby
RSpec.describe Campaigns::CloneInteractor do
  let :campaign_location do
    create(:campaign_location, \
           %i[allowed disallowed].sample,
           created_at: campaign_location_created_at)
  end

  pending 'add something'
end
```

2.

```ruby
RSpec.describe 'POST /api/v1/rewards' do
  def make_request
    post send(:"user_#{provider_name}_omniauth_callback_path"), \
         params: {},
         headers: { 'ACCEPT' => 'text/html', 'HTTP_JWT_AUD' => AllowlistedJwt::AADS_AUDIENCE }
  end

  pending 'add something'
end
```

**Comment by Earlopain** [comment]
Can you make a new issue please?
